src/logcounter.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2021 Staz Modrzynski
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "logcounter.hpp"
+
+rtc::LogCounter::LogCounter(plog::Severity severity, const std::string &text, std::chrono::seconds duration) :
+        severity(severity), text(text), duration(duration) {}
+
+rtc::LogCounter& rtc::LogCounter::operator++(int) {
+    std::lock_guard lock(mutex);
+    count++;
+    if (!future) {
+        future = ThreadPool::Instance().schedule(duration, [this]() {
+            int countCopy;
+            {
+                std::lock_guard lock(mutex);
+                countCopy = count;
+                count = 0;
```

Comment:
If you make `count` a `<std::atomic<int>`, you can replace this by an atomic exchange and you don't need the mutex anymore.

Reply:
Wouldn't it be possible for there to be a race between incrementing and testing the count? Based on your previous recommendation, and depending on the implementation, I could skip an enqueue or queue multiple times at the same time.

Reply:
If `count` is atomic, the increment operators are also atomic, so `if(count++ == 0)` increments the count and tests it without a possible race condition.

Reply:
Thanks for teaching my modern C++ haha